### PR TITLE
feat: Integrate zap widget

### PR DESCRIPTION
**Pull request by memoyil**
---

## PR-Codex overview
This PR introduces a new `CurrencyField` enum and updates various files to use it instead of `Field`.

### Detailed summary
- Added `CurrencyField` enum
- Replaced usages of `Field` with `CurrencyField` in multiple files

> The following files were skipped due to too many changes: `packages/localization/src/config/translations.json`, `apps/web/src/config/constants/zapV3.ts`, `apps/web/src/hooks/v3/useV3DerivedInfo.ts`, `apps/web/src/components/ZapLiquidityWidget.tsx`, `apps/web/src/views/AddLiquidityV3/formViews/V3FormView/index.tsx`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/views/AddLiquidityV3/ZapLiquidityWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            onTxSubmit={handleTransaction}
+            source="zap-widget"
+          />
+        </ModalContainer>
+      </ModalV2>
+    </>
+  )
+}
+
+export default ZapLiquidityWidget
```
Comment (chefjackson) [inline apps/web/src/views/AddLiquidityV3/ZapLiquidityWidget.tsx:116]: Let's use named export

File: apps/web/src/views/AddLiquidityV3/formViews/V3FormView/index.tsx [hunk 2]
```diff
@@ -570,6 +573,15 @@ export default function V3FormView({
           />
         </LockedDeposit>
       </DynamicSection>
+      {hasInsufficentBalance && (
+        <ZapLiquidityWidget
```
Comment (chefjackson) [inline apps/web/src/views/AddLiquidityV3/formViews/V3FormView/index.tsx:577]: Can we lazy load the widget? Otherwise it will impact our bundle size

**Comment by chefjackson** [comment]
@SocketSecurity ignore npm/react-redux@8.1.3
@SocketSecurity ignore npm/@reach/portal@0.18.0
@SocketSecurity ignore npm/@reach/utils@0.18.0

**Review by ChefJerry — DISMISSED** [review]

**Comment by chefjackson** [comment]
@SocketSecurity ignore npm/ws@7.4.6
@SocketSecurity ignore npm/debug@4.3.4
@SocketSecurity ignore npm/styled-jsx@5.1.1
@SocketSecurity ignore npm/postcss-load-config@3.1.4
@SocketSecurity ignore npm/vite-tsconfig-paths@4.2.1
@SocketSecurity ignore npm/vite@4.3.9
@SocketSecurity ignore npm/postcss-load-config@4.0.2
@SocketSecurity ignore npm/vite@5.0.12
@SocketSecurity ignore npm/ws@8.17.1

**Review by chefjackson — APPROVED** [review]

**Comment by chefjackson** [comment]
@SocketSecurity ignore npm/which@2.0.2
@SocketSecurity ignore npm/argparse@2.0.1
@SocketSecurity ignore npm/chalk@2.4.2
@SocketSecurity ignore npm/supports-color@5.5.0
@SocketSecurity ignore npm/resolve-from@4.0.0
@SocketSecurity ignore npm/path-exists@4.0.0
@SocketSecurity ignore npm/cross-spawn@7.0.3
@SocketSecurity ignore npm/isexe@2.0.0
@SocketSecurity ignore npm/braces@3.0.2
@SocketSecurity ignore npm/path-type@4.0.0
@SocketSecurity ignore npm/picocolors@1.0.0
@SocketSecurity ignore npm/locate-path@6.0.0
@SocketSecurity ignore npm/loose-envify@1.4.0
@SocketSecurity ignore npm/react-refresh@0.14.0
@SocketSecurity ignore npm/fast-glob@3.3.2

**Pull request merged**